src/main/java/org/killbill/billing/plugin/analytics/http/ReportsResource.java

```diff
@@ -259,7 +259,7 @@ public Result doPut(@Named("reportName") final String reportName,
         if (Boolean.TRUE.equals(shouldRefresh.orElse(Boolean.FALSE))) {
             reportsUserApi.refreshReport(reportName, context);
         } else {
-            reportsUserApi.updateReport(reportName, reportConfigurationJson, context);
+            reportsUserApi.updateReport(reportName, reportConfigurationJson.orElse(null), context);
```

Comment:
Fixed. The code now invokes the `reportsUserApi.updateReport` method only if a  `reportConfigurationJson` is present, otherwise, it returns a `BAD_REQUEST` status code.  Let me know if you would like this to be handled in some other way. 

If this looks good, I will add similar logic in the `doPost` method that handles report creation. Currently, if the `Create Report` endpoint is invoked without a report body, an exception is thrown.